runtime/sqlite_record_logger.hpp

```diff
@@ -104,7 +105,9 @@ class sqlite_thread {
         std::vector<record> record_batch{max_record_batch_size};
         std::size_t         actual_batch_size;
 
-        std::cout << "thread," << std::this_thread::get_id() << ",sqlite thread," << table_name << std::endl;
+        spdlog::get("illixr")->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [%n] [%^%l%$] [sqlite_record_logger] thread %t %v");
```

Comment:
Is there a better way to handle patterns here? Since these operations are not atomic, this might effect other calls to the global logger.

Reply:
Also, what would be the default format for the global logger?

Reply:
This is moot for this particular file since runtime/sqlite_record_logger.hpp is removed by the redo_timing-2 PR, but set_pattern() is also used in common/switchboard.hpp. In both cases, this was done in order to use the thread ID format specifier of %t. The motivation was to mimic the usage of the pre-spdlog output, which printed the thread IDs. Whether that's something y'all actually want and find useful is up to y'all.

As far as atomicity goes, I'm unsure. This is used in swtichboard::topic_subscription::thread_on_start(), how likely is this to cause a problem? Are threads getting created (and destroyed) continuously? From my limited experience, it doesn't seem that way.

The default format is essentially the same, minus the %t usage.

Reply:
I don't see anything in the spdlog wiki about thread safety regarding set_pattern(). The actual log sinks are made with the multithreaded-safe factories, so I think this is probably okay to do.